checker/src/visitors.rs

```diff
@@ -2198,18 +2226,24 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
     }
 
     /// Read the discriminant of an ADT and assign to path.
-    ///
-    /// Undefined (i.e. no effort is made to make it defined, but there’s no reason why it cannot
-    /// be defined to return, say, a 0) if ADT is not an enum.
     fn visit_discriminant(&mut self, path: Path, place: &mir::Place<'tcx>) {
         debug!(
             "default visit_discriminant(path: {:?}, place: {:?})",
             path, place
         );
-        let adtd_path = self.visit_place(place);
-        let adtd_type = self.get_place_type(place);
-        let adtd_value = self.lookup_path_and_refine_result(adtd_path, adtd_type);
-        self.current_environment.update_value_at(path, adtd_value);
+        let adt_path = self.visit_place(place);
+        let adt_path_length = adt_path.path_length();
+        assume!(adt_path_length < 1_000_000_000);
```

Comment:
This magic number could be a well-defined constant?

Reply:
It is pretty arbitrary. It just has to be big. The numeric literal is a good a name for it as any.